src/memory_service/fam_memory_service_direct.cpp

```diff
@@ -107,7 +109,10 @@ Fam_Memory_Service_Direct::Fam_Memory_Service_Direct(
         (libfabricProvider == NULL ? config_options["provider"].c_str()
                                    : libfabricProvider);
 
-    allocator = new Memserver_Allocator();
+    fam_path =
+        (fam_path == NULL ? config_options["fam_path"].c_str() : fam_path);
```

Comment:
fam_path is a local variable and being defined first time in this place. 
fam_path is anyway "NULL" at this place, so you can directly assign the value from config_options.